src/frequenz/sdk/actor/power_distributing/_battery_pool_status.py

```diff
@@ -0,0 +1,126 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+"""Class that stores pool of batteries and manage them."""
+
+import asyncio
+import logging
+from asyncio import ALL_COMPLETED
+from typing import Set
+
+from ...microgrid._battery import BatteryStatus, StatusTracker
+from .result import PartialFailure, Result, Success
+
+_logger = logging.getLogger(__name__)
+
+
+class BatteryPoolStatus:
+    """Holds pool of batteries and returns data from them."""
+
+    def __init__(
+        self,
+        battery_ids: Set[int],
+        max_data_age_sec: float,
+        max_blocking_duration_sec: float,
+    ) -> None:
+        """Create partially initialized object instance.
+
+        Note:
+            Please call `async_init` method to fully initialize BatteryPoolStatus. Otherwise
+            it is not possible to use BatteryPoolStatus.
+
+        Args:
+            battery_ids: set of batteries ids that should be stored in pool.
+            max_data_age_sec: If component stopped sending data, then
+                this is the maximum time when its last message should be considered as
+                valid. After that time, component won't be used until it starts sending
+                data.
+            max_blocking_duration_sec: This value tell what should be the maximum
+                timeout used for blocking failing component.
+        """
+        self._batteries = {
+            id: StatusTracker(id, max_data_age_sec, max_blocking_duration_sec)
+            for id in battery_ids
+        }
+        self._init_method_called: bool = False
+
+    async def async_init(self) -> None:
+        """Init battery pool."""
+        tasks = [
+            asyncio.create_task(battery.async_init())
+            for battery in self._batteries.values()
+        ]
+        await asyncio.wait(tasks, return_when=ALL_COMPLETED)
```

Comment:
Minor: I guess it could be jusut `asynio.gather()`, right?

Reply:
Done